packages/server-core/src/social/location/location.hooks.ts

```diff
@@ -210,23 +213,49 @@ export default {
     find: [discardQuery('action'), discardQuery('studio'), sortByLocationSetting],
     get: [],
     create: [
-      iff(isProvider('external'), verifyScope('location', 'write')),
+      when(
+        isProvider('external'),
+        iff(
+          isAction('studio'),
+          verifyScope('editor', 'write'),
+          resolveProjectId(),
+          verifyProjectPermission(['owner', 'editor'])
+        ).else(verifyScope('location', 'write'))
+      ),
       () => schemaHooks.validateData(locationDataValidator),
       schemaHooks.resolveData(locationDataResolver),
       persistData,
-      discard('locationSetting', 'locationAdmin')
+      discard('locationSetting', 'locationAdmin'),
+      discard('action')
     ],
     update: [disallow()],
     patch: [
-      iff(isProvider('external'), verifyScope('location', 'write')),
+      when(
+        isProvider('external'),
+        iff(
+          isAction('studio'),
+          verifyScope('editor', 'write'),
+          resolveProjectId(),
+          verifyProjectPermission(['owner', 'editor'])
+        ).else(verifyScope('location', 'write'))
+      ),
       () => schemaHooks.validateData(locationPatchValidator),
       schemaHooks.resolveData(locationPatchResolver),
       disallowNonId,
       persistData,
-      discard('locationSetting')
+      discard('locationSetting'),
+      discard('action')
     ],
     remove: [
-      iff(isProvider('external'), verifyScope('location', 'write')),
+      when(
+        isProvider('external'),
+        iff(
```

Comment:
@HexaField it seems like we need to make projectId part of location in core engine instead of MT. We did initially in https://github.com/EtherealEngine/etherealengine/pull/10123

As in this scenario, it seems like we need projectId to check for project permissions. Passing it from client which is not desired.

Ref:
`packages/projects/projects/eepro-multitenancy/services/hooks/social/location.hooks.ts`